server/model/monitor.js

```diff
@@ -512,10 +512,22 @@ class Monitor extends BeanModel {
                         }
                     }
 
-                    let tlsInfo;
-                    // Store tlsInfo when key material is received
-                    options.httpsAgent.on("keylog", (line, tlsSocket) => {
-                        tlsInfo = checkCertificate(tlsSocket);
+                    let tlsInfo = {};
+                    // Store tlsInfo when secureConnect event is emitted
+                    // The keylog event listener is a workaround to access the tlsSocket
+                    options.httpsAgent.once("keylog", async (line, tlsSocket) => {
```

Comment:
good catch on the `on` vs `once` difference.
Did not know said function. We are likely misusing `on` in other places as well

Reply:
Shouldn't be a big problem, the socket handlers do need `on` since they are repeatedly handling messages.